src/views/Layout.jsx

```diff
@@ -2,11 +2,31 @@ import { Outlet } from 'react-router-dom';
 
 import './Layout.css';
 import { auth } from '../api/config.js';
-import { useAuth, SignInButton, SignOutButton } from '../api/useAuth.jsx';
+import { useAuth } from '../api/useAuth.jsx';
 import LoggedOut from '../components/LoggedOut.jsx';
+import { useState } from 'react';
+import Button from '../components/Button.jsx';
+import { GoogleAuthProvider, signInWithPopup } from 'firebase/auth';
 
 export function Layout() {
+	const initialWindowSize = window.innerWidth;
+
 	const { user } = useAuth();
+	const [isMobile, setIsMobile] = useState(
```

Comment:
wanted to provide a note here, you can use `media queries` in css to accomplish this. I would suggest avoiding JS solutions like this.